src/api/artifact/controller.go

```diff
@@ -374,8 +377,28 @@ func (c *controller) assembleTag(ctx context.Context, tag *tm.Tag, option *TagOp
 		return t
 	}
 	if option.WithImmutableStatus {
-		// TODO populate immutable status
+		repo, err := c.repoMgr.Get(ctx, tag.RepositoryID)
+		if err != nil {
+			log.Error(err)
```

Comment:
Set the immutable to true when get erorr in case it is deleted by mistake?

Reply:
it's the original design, do we need to change?